pkg/controller/groupresourcequotaenforcer/groupresourcequotaenforcer_controller.go

```diff
@@ -0,0 +1,894 @@
+//
+// Copyright 2020 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package groupresourcequotaenforcer
+
+import (
+	"context"
+
+	// IBMDEV
+	"reflect"
+
+	admissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	operatorv1alpha1 "github.ibm.com/IBMPrivateCloud/ibm-mt-grqenforcer-operator/pkg/apis/operator/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_groupresourcequotaenforcer")
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+/*
+IBMDEV
+This type and global variable provide the names for all the resources created/managed by this operator.
+As such, it provides an overview of the required pieces for the enforcement component.
+Note however that the certSecret isn't currently used (secret name is provided by CR param instead).
+*/
+type grqeNameSuffix struct {
+	//certSecret       string	/*currently unused, see commented out code blocks*/
+	grqeDeployment     string
+	grqeService        string
+	grqeWebhook        string
+	bridgeDeployment   string
+	bridgeService      string
+	serviceAccount     string
+	clusterRole        string
+	clusterRoleBinding string
+}
+
+var suffix grqeNameSuffix = grqeNameSuffix{
```

Comment:
The go linter complains that type `grqeNameSuffix` here is unnecessary, it will be inferred from the right-hand side.